addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+        }
+        self._l10n_tw_edi_prepare_item_list(json_data)
+
+        if self.l10n_tw_edi_is_donate:
+            json_data.update({"LoveCode": self.l10n_tw_edi_love_code})
+        return json_data
+
+    def _l10n_tw_edi_send(self, json_content):
+        request_url, merchant_id, hashkey, hashIV = self.company_id._l10n_tw_edi_prepare_api_param()
+        # Ensure to lock the records that will be sent, to avoid risking sending them twice.
+        self.env["res.company"]._with_locked_records(self)
+        # prepare EcPayAPI class
+        ecpay_api = EcPayAPI(hashkey, hashIV, merchant_id)
+        response_data = ecpay_api.call_ecpay_api(request_url + "/Issue", json_content)
+        if response_data.get("RtnCode") != 1:
+            return [_("Invoice '%(name)s' Error: '%(error_message)s'") % {"name": self.name, "error_message": response_data.get("RtnMsg")}]
+        self.l10n_tw_edi_ecpay_invoice_id = response_data.get("InvoiceNo")
+        self.l10n_tw_edi_invoice_create_date = ecpay_api._transfer_time(response_data.get("InvoiceDate").replace("+", " "))
+        self.l10n_tw_edi_state = "invoiced"
+        self._message_log(
+            body=_("The invoice has been successfully sent to Ecpay with Ecpay invoice number '%(invoice_number)s'.") % {'invoice_number': response_data.get('InvoiceNo')},
+        )
+
+    def _l10n_tw_edi_fetch_ecpay_invoice(self):
+        request_url, merchant_id, hashkey, hashIV = self.company_id._l10n_tw_edi_prepare_api_param()
+        # Ensure to lock the records that will be sent, to avoid risking sending them twice.
+        self.env["res.company"]._with_locked_records(self)
+        # prepare json data
+        json_data = {
+            "MerchantID": merchant_id,
+            "RelateNumber": self.l10n_tw_edi_related_number,
+        }
+
+        # prepare EcPayAPI class
+        response_data = EcPayAPI(hashkey, hashIV, merchant_id).call_ecpay_api(request_url + "/GetIssue", json_data)
+        if response_data.get('RtnCode') != 1:
+            return [_("Invoice '%(name)s' Error: '%(error_message)s'") % {"name": self.name, "error_message": response_data.get("RtnMsg")}]
+
+        self.write({
+            "l10n_tw_edi_invoice_amount": response_data.get("IIS_Sales_Amount"),
+            "l10n_tw_edi_invoice_shop_custom_number": response_data.get("IIS_Relate_Number"),
+            "l10n_tw_edi_remain_refundable_amount": response_data.get("IIS_Remain_Allowance_Amt"),
+            "l10n_tw_edi_state": "valid" if response_data.get("IIS_Invalid_Status") == "0" else "invalid",
+        })
+
+    def _l10n_tw_edi_run_invoice_invalid(self):
+        if self.l10n_tw_edi_ecpay_invoice_id:
+            if self.l10n_tw_edi_state == "invalid":
+                raise UserError(_("The following invoice: '%(invoice_id)s' is abolished") % {'invoice_id': self.l10n_tw_edi_ecpay_invoice_id})
+        else:
+            raise UserError(_("Cannot find related invoice"))
```

Comment:
Same 
```suggestion
            raise UserError(_("You cannot invalidate an invoice that was not sent to Ecpay."))
```
I'll check the error message more carefully in a later review